File: apps/api/Finly.Domain/Entities/Occurrence.cs
using Finly.Domain.Common;
using Finly.Domain.Enums;

namespace Finly.Domain.Entities;

public class Occurrence : BaseEntity
{
    public Guid TransactionId { get; set; }

    public int? InstallmentIndex { get; set; }
    public DateOnly DueDate { get; set; }
    public decimal Amount { get; set; }
    public OccurrenceStatus Status { get; set; }
    public DateTime? PaidAt { get; set; }
    public bool IsCustomized { get; set; }

    public Transaction Transaction { get; set; } = null!;
}

File: apps/web/src/types/occurrence.ts
export type OccurrenceStatus = "pending" | "paid";

export type Occurrence = {
  id: string;
  transactionId: string;
  installmentIndex: number | null;
  dueDate: string;
  amount: number;
  status: OccurrenceStatus;
  paidAt: string | null;
  isCustomized: boolean;
  createdAt: string;
};
